### ISSUE TITLE
search is not great

### ISSUE THREAD

**Issue by 89z** [rust-lang/crates.io#5222]
If you search for `http`:

https://crates.io/search?q=http

you don't find `reqwest`:

https://crates.io/crates/reqwest

personally I don't care for Reqwest, but my understanding is its one of, if not the most popular crate for HTTP client and similar. I checked the first 10 pages, and its not listed at all. It has:

- 36 million total downloads
- 5 million recent downloads
- HTTP in description
- and the `http-client` category
- recently updated (3 months)
- using the `http` keyword

in short, its doing everything I can think of to scream "HEY, LOOK AT ME", but search doesn't find it at all. The default Sort by is Relevance, does that just care about the name only?

**Comment by Turbo87** [comment]
the search is looking at multiple fields to determine the relevance, but yes, it does not include download numbers and such things in the metric.

FWIW there are already a few other issues open about this so I'll go ahead and close this as a duplicate. If you have ideas on how to improve the situation then help would be greatly appreciated.

**Comment by 89z** [comment]
I am happy to try to help with this. Can you please point to a single other issue that this is a duplicate of? I am not familiar with this tracker, so thats kind of a vague comment you made.

**Comment by carols10cents** [comment]
https://github.com/rust-lang/crates.io/issues?q=is%3Aopen+label%3AA-search+sort%3Aupdated-desc

**Comment by 89z** [comment]
I really don't think this is a duplicate of any other issue. I linked one that pretty close, but I think that issue should be closed in favor of this one, not the other way around.

The problem is not JUST the keywords. Even ignoring the keyword factor, the crate I gave as example has a HUGE number of downloads, both total and recent. So downloads are also under valued in a search, if they are valued at all.
